tools/release/release.py

```diff
@@ -123,17 +123,20 @@ def build_package(cfg: Config, pkg_path: str, build_dir: str) -> None:
 
 def _next_version(version: str) -> str:
     cur = parse(version)
+    assert isinstance(cur, Version)
     if cur.is_devrelease:
         prefix = "dev"
+        assert cur.dev is not None
         num = cur.dev + 1
         new_version = f"{cur.major}.{cur.minor}.{cur.micro}.{prefix}{num}"
     elif cur.is_prerelease:
+        assert cur.pre is not None
         prefix = cur.pre[0]
         num = cur.pre[1] + 1
         new_version = f"{cur.major}.{cur.minor}.{cur.micro}.{prefix}{num}"
     elif cur.is_postrelease:
-        prefix = cur.post[0]
-        num = cur.post[1] + 1
+        assert cur.post is not None
+        num = cur.post + 1
         new_version = f"{cur.major}.{cur.minor}.{cur.micro}.{prefix}{num}"
```

Comment:
This looks a bit strange. prefix is mentioned in the f string, but no longer set?
Also why change cur.post[1] to cur.post ?

Reply:
> Also why change cur.post[1] to cur.post?

The type of `cur.post` is `Optional[int]`, so `cur.post[1]` does not make sense.
As for why the previous version had `cur.post[0]` and `cur.post[1]`: I suspect that this branch of the if-statement was copy/pasted from the `if cur.is_prerelease` branch above.

There is currently no test coverage for this `release.py` file.

Reply:
> This looks a bit strange. prefix is mentioned in the f string, but no longer set?

Updated in https://github.com/facebookresearch/hydra/pull/1878/commits/f0d30ae370a0cd2faed886fcba7070ee91a40fa9. Thanks!